yi-core/src/Yi/Tag.hs

```diff
@@ -120,21 +120,20 @@ importTagTable filename = do
   return TagTable { tagFileName = takeFileName filename
                   , tagBaseDir  = takeDirectory filename
                   , tagFileMap  = cts
-                    -- TODO either change word-trie to use Text or
-                    -- figure out a better way all together for this
-                  , tagTrie = Trie.fromList . map (T.unpack . _unTag) $ keys cts
+                  , tagCompletionTree = CT.fromList . map (_unTag) $ keys cts
                   }
 
 -- | Gives all the possible expanded tags that could match a given @prefix@
 hintTags :: TagTable -> T.Text -> [T.Text]
-hintTags tags prefix = map (T.append prefix . T.pack) sufs
+hintTags tags prefix = map (prefix `T.append`) sufs
```

Comment:
I think there should be "higher" level functions like this in other modules like `Yi.TextCompletion` and `Yi.Completion`.